plugins/com.aptana.js.core/src/com/aptana/js/internal/core/node/NodePackageManager.java

```diff
@@ -491,14 +491,21 @@ public IPath getModulesPath(String packageName, boolean isGlobal, String... args
 		}
 
 		IStatus status = runInBackground(CollectionsUtil.toArray(processArgs));
-		if (!status.isOK())
+		IPath moduleIPath = null;
+		boolean isPathExists = false;
+		String message = status.getMessage();
+		if (!StringUtil.isEmpty(message))
+		{
+			String[] lines = message.split("\n"); //$NON-NLS-1$
+			moduleIPath = Path.fromOSString(lines[lines.length - 1]);
+			isPathExists = moduleIPath.toFile().exists();
+		}
+		if (!status.isOK() && !isPathExists) // some cases status.isNotOK even though it finds valid path
 		{
 			throw new CoreException(new Status(IStatus.ERROR, JSCorePlugin.PLUGIN_ID, MessageFormat.format(
 					Messages.NodePackageManager_FailedListPackageError, packageName)));
```

Comment:
We are not logging the error message anywhere from the Status object. We are just creating a new Status object and wrapping it up in CoreException to return the error. That way, we will scratch our head again if some user reports this error again, and we can't see any useful information in the logs.

So, it should be a good practice to always log errors and return whatever information/error we need to.